### PR TITLE
Stop removing GpuCoalesceBatches from non-AQE queries when AQE is enabled

### PR DESCRIPTION
**Pull request by andygrove**
Signed-off-by: Andy Grove <andygrove@nvidia.com>

`GpuTransitionOverrides` has special handling for adaptive queries, where `GpuCoalesceBatches` is removed from a `GpuShuffleExchangeExec` and re-inserted around the `GpuCustomShuffleReader`. See comments in this PR for a more detailed explanation.

The bug here though was that we assumed all queries where adaptive queries when AQE is enabled, and there are cases where queries are not adaptive (such as when dynamic partition pruning is used) and this resulted in the plugin removing the `GpuCoalesceBatches` operator and not inserting it again (because there was no `GpuCustomShuffleReader`) and this results in poor performance.

This closes https://github.com/NVIDIA/spark-rapids/issues/698

### CODE REVIEW

**Comment by andygrove** [comment]
build

**Comment by andygrove** [comment]
build

**Comment by andygrove** [comment]
The scala tests failed this time against spark 3.0.1 and I cannot reproduce locally.

**Comment by andygrove** [comment]
build

**Comment by andygrove** [comment]
build

File: docs/configs.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ Name | Description | Default Value
...
 <a name="sql.format.parquet.multiThreadedRead.maxNumFilesParallel"></a>spark.rapids.sql.format.parquet.multiThreadedRead.maxNumFilesParallel|A limit on the maximum number of files per task processed in parallel on the CPU side before the file is sent to the GPU. This affects the amount of host memory used when reading the files in parallel.|2147483647
-<a name="sql.format.parquet.multiThreadedRead.numThreads"></a>spark.rapids.sql.format.parquet.multiThreadedRead.numThreads|The maximum number of threads, on the executor, to use for reading small parquet files in parallel.|20
+<a name="sql.format.parquet.multiThreadedRead.numThreads"></a>spark.rapids.sql.format.parquet.multiThreadedRead.numThreads|The maximum number of threads, on the executor, to use for reading small parquet files in parallel. This can not be changed at runtime after the executor hasstarted.|20
```
Comment (jlowe) [inline docs/configs.md:60]: This indicates the branch needs to be upmerged/rebased on latest branch-0.2.

**Comment by andygrove** [comment]
build

**Comment by andygrove** [comment]
Status update: I have manually tested this by running TPC-DS benchmarks and confirming no regressions in performance (with our derived TPC-DS benchmarks, so no DPP involved).

**Review by jlowe — APPROVED** [review]

**Review by revans2 — APPROVED** [review]

**Pull request merged**